tests/cpp/unit-tests/CMakeLists.txt

```diff
@@ -65,6 +67,36 @@ set(UNIT_TESTS
     test_redisserver.cpp
 )
 
+# Compare the number of source files given above to the number of source CPP
+# files in the directory. If the two numbers aren't equal, we're not testing
+# coverate completely. It is not recommended to use globbed file lists
+# in add_executable()
+list(LENGTH SOURCES source_count)
+
+file(GLOB CPP_SOURCE_FILES ../../../src/cpp/*.cpp)
+
+list(LENGTH CPP_SOURCE_FILES cpp_source_file_count)
+
+if(NOT (source_count EQUAL cpp_source_file_count))
+	message(FATAL_ERROR "CPP test coverage sources count (${source_count}) \
+does not match count of cpp source files: ${cpp_source_file_count}")
+endif()
+
+# Compare the number of unit test files given above to the number of unit
+# test CPP files in the directory. If the two numbers aren't equal, we're
+# not running all the tests! It is not recommended to use globbed file lists
+# in add_executable()
+list(LENGTH UNIT_TESTS unit_test_count)
+
+file(GLOB UNIT_TEST_FILES *.cpp)
+
+list(LENGTH UNIT_TEST_FILES cpp_unit_test_file_count)
+
+if(NOT (unit_test_count EQUAL cpp_unit_test_file_count))
+	message(FATAL_ERROR "CPP test coverage sources count (${unit_test_count}) \
```

Comment:
We should probably have a different error message than the sources message. Make sure this message references test file count rather than cpp source file count. e.g. "Target unit test file count does not match actual unit test file count"